jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/Histogram.java

```diff
@@ -0,0 +1,245 @@
+package pl.edu.uj.JImageStream.filters;
+
+import java.awt.Color;
+import java.awt.image.BufferedImage;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import pl.edu.uj.JImageStream.api.core.Filter;
+import pl.edu.uj.JImageStream.model.ColorChannel;
+import pl.edu.uj.JImageStream.model.Pixel;
+
+public class Histogram extends Filter{
+
+    private int[][] histogram = new int[4][256]; // 0=R, 1=G, 2=B, 3=A
+    private int[] blackAndWhiteHistogram = new int[256];
+    private boolean[] channel = new boolean[4]; // 0=R, 1=G, 2=B, 3=A
+    private ColorChannel[] channels;
+    private final int TOPMARGIN = 0;
+    private BufferedImage image = null;
+    private BufferedImage output = null;
+    private int outputWidth;
+    private int outputHeight;
+    private boolean readyToApply;
+    private boolean blackAndWhite;
```

Comment:
try to group variables with each other, it's hard to understand what is necessary in blocks.